docs/design/adrs/2024-05-02-enable-project-versioning

```diff
@@ -0,0 +1,63 @@
+# Use Github Releases to manage project versioning
```

Comment:
Thanks @frtibble for coming up with this. Some thoughts:
- I'm interested to understand why the proposal is to make the release a manual process when it is possible to automate via GitHub actions? 
- So far we've been releasing each change into the main branch on a green build. Before we decide on changing this to a less frequent cadence, it would be good to understand why we can't continue to release every PR on green when we use versioning. (Releasing smaller changesets is preferred in general since it is less riskier and easier to troubleshoot as compared to collecting changes for a week and releasing everything together)
- Not sure if it is part of this ADR or a new one, but it would help to create some guidelines around what we consider breaking changes for this repo, for example - change in search index schema etc.

Reply:
Thanks @superhindupur for your comments!

My suggestion for manual releases was mostly from reviewing some larger project repositories and looking at their release cadences. It looked to me like they were bundling multiple changes together and releasing e.g. on a weekly basis. I didn't see any automated workflows so thought this might be a preferred approach, to review the changes on a less frequent basis and publish a release manually.

That said, your point on the current release cadence made me rethink this suggestion. I imagine those larger projects have longer build/release/test timelines, which forces bundling of changes together. If we are currently releasing on every PR on green, then the most logical change I think would be to simply add versioning to that, unless there was strong reason to change the cadence also. But I don't think adding versioning impacts that, hence the update to the proposal in this ADR 😊

And on guidelines - great suggestion, I think this should be incorporate in the work for this as a sub-task, so have included that also!